gno.land/cmd/gnoland/config_edit.go

```diff
@@ -0,0 +1,264 @@
+	// ex: x.y.field
+	if len(pathParts) > 1 {
+		// Get the remaining path
+		remainingPath := strings.Join(pathParts[1:], ".")
+
+		// Recursively try to traverse the path and update the given field
+		return updateFieldAtPath(field, remainingPath, value)
+	}
+
+	// We've reached the actual field, check if it can be updated
+	if !field.CanSet() {
+		return fmt.Errorf("unable to set value")
+	}
+
+	// Convert the value to the field's type
+	v, err := convertStringToType(value, field.Interface())
+	if err != nil {
+		return fmt.Errorf("unable to convert value to field type, %w", err)
+	}
+
+	// Update the field value
+	field.Set(reflect.ValueOf(v))
+
+	return nil
+}
+
+// convertStringToType attempts to convert the given
+// string value to an output type.
+// Because we opted to using reflect instead of a flag-based approach,
+// arguments (always strings) need to be converted to the field's
+// respective type, if possible
+func convertStringToType(value string, outputType any) (any, error) {
+	parseInt := func(size int) (any, error) {
+		castValue, err := strconv.ParseInt(value, 10, size)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert to int%d, %w", size, err)
+		}
+
+		return castValue, nil
+	}
+
+	parseUint := func(size int) (any, error) {
+		castValue, err := strconv.ParseInt(value, 10, size)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert to uint%d, %w", size, err)
+		}
+
+		return castValue, nil
+	}
+
+	parseFloat := func(size int) (any, error) {
+		castValue, err := strconv.ParseFloat(value, size)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert to float%d, %w", size, err)
+		}
+
+		return castValue, nil
+	}
+
+	switch outputType.(type) {
+	case string:
+		return value, nil
+	case []string:
+		// This is a special case.
+		// Since values are given as a single string (argument),
+		// they need to be parsed from a custom format.
+		// In this case, the format for a []string is comma separated:
+		// value1,value2,value3 ...
+		return strings.SplitN(value, ",", -1), nil
+	case time.Duration:
+		castValue, err := time.ParseDuration(value)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert to Duration, %w", err)
+		}
+
+		return castValue, nil
+	case int:
+		castValue, err := strconv.Atoi(value)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert to int, %w", err)
+		}
+
+		return castValue, nil
+	case int8:
+		return parseInt(8)
+	case int16:
+		return parseInt(16)
+	case int32:
+		return parseInt(32)
+	case int64:
+		return parseInt(64)
+	case bool:
+		castValue, err := strconv.ParseBool(value)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert to bool, %w", err)
+		}
+
+		return castValue, nil
+	case uint, uint64:
+		return parseUint(64)
+	case uint8:
+		return parseUint(8)
+	case uint16:
+		return parseUint(16)
+	case uint32:
+		return parseUint(32)
+	case float32:
+		return parseFloat(32)
+	case float64:
+		return parseFloat(64)
```

Comment:
```suggestion
	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64. bool:
		v := reflect.New(reflect.TypeOf(outputType))
		if err := json.Unmarshal([]byte(value), v.Interface()); err != nil {
			return nil, err
		}
		return v.Interface(), nil
```

And remove parseInt / parseUint. I don't think they're working correctly anyway, if you want I can elaborate further. But this is a simple reflect-based parser that exists 800 times elsewhere; let's re-use one of the most used implementations in Go.

Reply:
This suggestion doesn't work for `int` values and `bool` types, you can try running the testing suite against this change:

ex:
```shell
panic: reflect.Set: value of type *int is not assignable to type int [recovered]
	panic: reflect.Set: value of type *int is not assignable to type int
```

Can you please elaborate what's the problem, or why the previous implementation wasn't sufficient?